edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/client/NamespaceKubernetesClientExtensions.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest.Client
+{
+    using System.Threading.Tasks;
+    using k8s;
+    using k8s.Models;
+
+    public static class NamespaceKubernetesClientExtensions
```

Comment:
why create extension method instead of adding a method in KubernetesClient class?

Reply:
I can foresee a lot of custom methods will be added here. So instead of putting everything in one class, we can group them by Resource name. There are multiple ways to do it, partial classes for example.
I decided to go with extension methods because it is somewhat pattern I saw in [c# k8s client](https://raw.githubusercontent.com/kubernetes-client/csharp/8ab95b6b920e8a10b4c0122e09d158be2005aa15/src/KubernetesClient/generated/KubernetesExtensions.cs). Because I indeed extend client functionality with additional methods and don't create a new composition of objects.

Reply:
I can see that in C# k8s client; they implemented the extension method for IKubernetes which is an interface; therefore it can't put implementation inside and interface.  Also since that interface is publicly released, they may not be easy to change it to abstract class.  I still see any benefit to implement this by extension methods.  If the class is getting bigger, that is a sign to refactor and break down the class as it may have too many responsibilities.